app/dagger/SimpleInjector.java

```diff
@@ -0,0 +1,67 @@
+package dagger;
+
+import play.api.inject.BindingKey;
+import play.inject.Injector;
+import scala.reflect.ClassTag;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * A simple injector with additional classes...
+ */
+public class SimpleInjector implements Injector {
```

Comment:
I'm not sure if that's any better -- we have a java injector() method that would call out to a scala one, and then convert it back to java...